astropy/stats/funcs.py

```diff
@@ -761,22 +773,44 @@ def median_absolute_deviation(a, axis=None):
     # See https://github.com/numpy/numpy/issues/7330 why using np.ma.median
     # for normal arrays should not be done (summary: np.ma.median always
     # returns an masked array even if the result should be scalar). (#4658)
-    if isinstance(a, np.ma.MaskedArray):
+    if isinstance(data, np.ma.MaskedArray):
         func = np.ma.median
+        if ignore_nan:
+            data = np.ma.masked_where(np.isnan(data), data)
```

Comment:
@mwcraig since you're digging into this, any chance you could put up a grid of which versions do what?  I'm a little lost about which numpy versions do/do not return NaNs (in)appropriately, and I'm working on other things so having a hard time keeping track of the differences.